IPython/core/prompts.py

```diff
@@ -20,7 +20,7 @@ def __str__(self):
         return str(self())
 
     def __unicode__(self):
-        return py3compat.unicode_type(self())
+        return self.__str__()
```

Comment:
The `__unicode__` magic method is no longer used on Python 3, so we can get rid of it.